lib/Zonemaster/CLI.pm

```diff
@@ -517,13 +518,24 @@ sub add_fake_delegation {
         if ($ip) {
             push @{ $data{ $self->to_idn( $name ) } }, $ip;
         }
+        elsif (Zonemaster::Zone->new( { name => $domain } )->is_in_zone( $name )) {
+            say STDERR "--ns in zone nameservers must be a name/ip pair.";
+            exit( 1 );
+        }
```

Comment:
In facts, this error is like others... CLI errors. It is consistent with the way the CLI was written. We can plan improvements. First we can check all args and just exit when all args has been verified, not only when it meets the first mistake. This work is only on the CLI part. Second, we can add this in the engine itself, it is more complicated because, as you noticed, the design must be modified.

If the user demand is to catch cases when a nameserver is not correct or need an IP because it is in zone, the first solution can be enough, at least at a first step.

What do you thing @aabdnn ?